tests/cli/test_project.py

```diff
@@ -4,11 +4,64 @@
 import pytest
 import readchar
 import yaml
+from prefect.server import models
+from prefect.client import schemas
 from test_cloud import interactive_console  # noqa
 
 from prefect.testing.cli import invoke_and_assert
 
 
+@pytest.fixture
+async def deployment_with_pull_step(
+    session,
+    flow,
+):
+    def hello(name: str):
+        pass
+
+    deployment = await models.deployments.create_deployment(
+        session=session,
+        deployment=schemas.objects.Deployment(
+            name="hello",
+            flow_id=flow.id,
+            pull_steps=[
+                {"prefect.projects.steps.set_working_directory": {"directory": "/tmp"}},
+            ],
+        ),
+    )
+    await session.commit()
+
+    return deployment
+
+
+@pytest.fixture
+async def deployment_with_pull_steps(
+    session,
+    flow,
+):
+    def hello(name: str):
+        pass
+
+    deployment = await models.deployments.create_deployment(
+        session=session,
+        deployment=schemas.objects.Deployment(
+            name="hello",
+            flow_id=flow.id,
+            pull_steps=[
+                {
+                    "prefect.projects.steps.set_working_directory": {
+                        "directory": "/tmp"
+                    },
+                },
+                {"prefect.projects.steps.set_working_directory": {"directory": "/tmp"}},
```

Comment:
Can you make the second `directory` different? That way, this will exhibit different behavior than the fixture above.

Reply:
Yeah, good idea